parsers/IL.py

```diff
@@ -88,19 +141,66 @@ def extract_price_date(soup):
 def fetch_production(
     zone_key="IL", session=None, target_datetime=None, logger=None
 ) -> dict:
-    if target_datetime:
-        raise NotImplementedError("This parser is not yet able to parse past dates")
 
-    data = fetch_all()
-    production = [float(item) for item in data]
-
-    # all mapped to unknown as there is no available breakdown
-    return {
-        "zoneKey": zone_key,
-        "datetime": arrow.now(TZ).datetime,
-        "production": {"unknown": production[0] + production[1]},
-        "source": IEC_URL,
-    }
+    data = {"zoneKey": zone_key, "datetime": arrow.now(TZ).datetime}
+
+    # fetch pv capacity from IRENA
+
+    try:
+        assert (
+            post(url=IRENA_CAPACITY_URL, json=IRENA_SOLAR_PV_QUERY).status_code == 200
+        ), "IRENA solar PV url is not accessible"
+        if target_datetime:
+            data["datetime"] = arrow.get(target_datetime).datetime
+        solar_pv = fetch_IRENA_capacity(
+            irena_url=IRENA_CAPACITY_URL, irena_query=IRENA_SOLAR_PV_QUERY
+        )
+
+    except AssertionError:
+        solar_pv = pd.DataFrame()
+
+    # fetch  solar thermal capacity from IRENA
+    try:
+        assert (
+            post(url=IRENA_CAPACITY_URL, json=IRENA_SOLAR_THERMAL_QUERY).status_code
+            == 200
+        ), "IRENA solar thermal url is not accessible"
+        if target_datetime:
+            data["datetime"] = arrow.get(target_datetime).datetime
+        solar_thermal = fetch_IRENA_capacity(
+            irena_url=IRENA_CAPACITY_URL, irena_query=IRENA_SOLAR_THERMAL_QUERY
+        )
+    except AssertionError:
+        solar_thermal = pd.DataFrame()
+
+    solar = pd.concat([solar_pv, solar_thermal]).groupby("year").sum()
+
+    # load solar capacity to data
+    try:
+        data["capacity"] = {"solar": solar.loc[data["datetime"].year, "capacity"]}
```

Comment:
That's going to fail if the provided `target_datetime` is not a `datetime` compatible object (ex: `str`)